### ISSUE TITLE
Better error messages for "... is not a valid function argument name"

### ISSUE THREAD

**Issue by CameronBieganek** [JuliaLang/julia#45031]
Is there any chance that we can get a better error message here? I encountered this when writing a unit test (I've probably encountered it before).

```julia
julia> foo(x) = x;

julia> bar(x) = x;

julia> a = 1;

julia> foo(bar(a)) == 42
false

julia> foo(bar(a)) = 42
ERROR: syntax: "bar(a)" is not a valid function argument name around REPL[40]:1
```

Related issue: #40354

**Comment by simeonschaub** [comment]
What do you think the error should say instead? Perhaps just adding "inside the function definition for \`foo`" at the end of the error message might already make things clearer?

**Comment by CameronBieganek** [comment]
Yeah, I think  "inside the function definition for \`foo\`" would definitely help. It would hopefully help the user to realize that `foo(bar(a)) = 42` is the single-line function definition syntax (aside from the invalid argument name) rather than an equality comparison.

**Comment by Moelf** [comment]
"perhaps you have `=` instead of `==`?"

**Comment by CameronBieganek** [comment]
> "perhaps you have `=` instead of `==`?"

Yeah, it does seem like that is the main way that this error pops up, so that would probably be a useful addition to the error message. Are there any other common cases that trigger this error?
